src/ext/java/org/opentripplanner/ext/siri/updater/SiriETGooglePubsubUpdater.java

```diff
@@ -290,6 +248,57 @@ private void addShutdownHook() {
     }
   }
 
+  private Subscription createSubscription() {
+    try (SubscriptionAdminClient subscriptionAdminClient = SubscriptionAdminClient.create()) {
+      return subscriptionAdminClient.createSubscription(
+        Subscription
+          .newBuilder()
+          .setTopic(topic.toString())
+          .setName(subscriptionName.toString())
+          .setPushConfig(pushConfig)
+          .setMessageRetentionDuration(
+            // How long will an unprocessed message be kept - minimum 10 minutes
+            com.google.protobuf.Duration.newBuilder().setSeconds(600).build()
+          )
+          .setExpirationPolicy(
+            ExpirationPolicy
+              .newBuilder()
+              // How long will the subscription exist when no longer in use - minimum 1 day
+              .setTtl(com.google.protobuf.Duration.newBuilder().setSeconds(86400).build())
+              .build()
+          )
+          .build()
+      );
+    } catch (IOException e) {
+      // Google libraries expects credentials json-file either as
+      //   Path is stored in environment variable "GOOGLE_APPLICATION_CREDENTIALS"
+      //   (https://cloud.google.com/docs/authentication/getting-started)
+      // or
+      //   Credentials are provided through "workload identity"
+      //   (https://cloud.google.com/kubernetes-engine/docs/concepts/workload-identity)
+      throw new RuntimeException(
+        "Unable to initialize Google Pubsub-updater: System.getenv('GOOGLE_APPLICATION_CREDENTIALS') = " +
+        System.getenv("GOOGLE_APPLICATION_CREDENTIALS")
+      );
+    }
+  }
+
+  private void deleteSubscription() {
+    try (SubscriptionAdminClient subscriptionAdminClient = SubscriptionAdminClient.create()) {
```

Comment:
I'm curious if you're really meant to create the client just for deletion. Is that correct?

Reply:
Yes, this is the admin client, not the client used to receive messages from the subscription.
The admin client is used only at startup to create the subscription and at shutdown to delete the subscription.